### DISCUSSION TITLE
Is there a proper way to upgrade ASP.NET boilerplate codebase that uses an older version?

### DISCUSSION THREAD

**Discussion by timematcher** [Dotnet-Boxed/Templates#167] [Q&A]
@RehanSaeed Great work so far. I have a question:

I started working on a couple of sites of my own a few months ago using the **3.x versions of the ASP.NET boilerplate**. During these past months, several new version of this template were released as part of the ongoing advancements on this template corresponding to the new releases of the .NET Core and .NET framework.

**My question is :** Is there a proper way to upgrade those old code bases to the new version of the template without causing much mess? 

I know its too much to ask for a an upgrade tool (as it will be really hard to develop, test and maintain such a tool and it will require a lot of time and effort with probably little value vs cost)..., **are there any steps one can follow to manually upgrade the code base from lets say, v4.x to v6.x of ASP.NET boilerplate?**

I know it is difficult question to answer and probably there is no answer to this question at all. I know its simply too much to ask! I will simply appreciate some pointers if there were a way to go about doing such an upgrade manually for case.

Thanks

**Comment by RehanSaeed** [comment]
This is a duplicate question. I already move a lot of code into NuGet packages which get updated. There is no real way to do this for template code as people change that code and customize it. Microsoft is going to do something in this space in their new templating engine but there are few details yet and I don't know how it will work. Best thing to do is look at the release notes and update your project.
